code/socialDistribution/views.py

```diff
@@ -1,13 +1,82 @@
 from django.shortcuts import render
+from django.contrib import messages
 from django.http import HttpResponse
-# Create your views here.
+from django.shortcuts import redirect
+from django.contrib.auth.models import Group
+from django.contrib.auth import authenticate, login, logout
+
+from .models import Author
+from .forms import CreateUserForm
+from .decorators import allowedUsers, unauthenticated_user
 
 def index(request):
     return HttpResponse("Hello, world. You're at the Login/SignUp Page.")
 
+@unauthenticated_user
+def loginPage(request):
+    """
+        Logs in a user and redirects to Home page
+    """
+    if request.method == 'POST':
+        username = request.POST.get('username')
+        password = request.POST.get('password')
+
+        user = authenticate(request, username=username, password=password)
+
+        if user is not None:
+            login(request, user)
+            return redirect('home')
+        else:
+            messages.info(request, "Username or Passoword is incorrect.")
+
+    return render(request, 'user/login.html')
+
+@unauthenticated_user
+def register(request):
+    """
+        Registers a new user and redirects to Login page
+    """
+    form = CreateUserForm()
+
+    if request.method == 'POST':
+        form = CreateUserForm(request.POST)
+        if form.is_valid():
+            user = form.save()
+            # extract form data
+            username = form.cleaned_data.get('username')
+            first_name = form.cleaned_data.get('first_name')
+            last_name = form.cleaned_data.get('last_name')
+            github_url = request.POST.get('github_url', '')
```

Comment:
Using an incorrect github url, I got a 500 error but the registration still went through. Should we do some error handling?

Reply:
Added GitHub URL check. Thanks.